legateboost/models/tree.py

```diff
@@ -46,13 +54,27 @@ def __init__(
         self,
         max_depth: int = 8,
         split_samples: int = 256,
-        alpha: float = 1.0,
```

Comment:
FWIW, I think it would make sense to use a `, *,` here (strictly speaking necessary to since passing by position now makes it the `l1_regularization` parameter). 

(Didn't revisit previous signatures)

And/or maybe just put `l2_reg` first always. since it is also the pattern in the C-code.